### PR TITLE
split resource tools into opentelemetry-propagator-gcp and opentelemetry-resourcedetector-gcp

### PR DESCRIPTION
**Pull request by aabmass**
Split `opentelemetry-tools-google-cloud` package into separate
- `opentelemetry-propagator-gcp`
- `opentelemetry-resourcedetector-gcp`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -58,18 +58,18 @@ jobs:
     steps:
       - setup
       - run_with_pip_cache:
-          cache_key_base: lint-python3.8
+          cache_key_base: lint-python3.9
           steps:
-            - run: tox -f lint-ci
+            - run: tox -s false -f lint-ci
```
Comment (punya) [inline .circleci/config.yml:63]: Nit: When writing scripts or defining CI steps, consider using long-form command line flags for readability (I had to look up that this meant `--skip-missing-interpreters`).

**Review by punya — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (punya) [reply .circleci/config.yml:63]: This is non-blocking feedback. There are lots of single-character flags used already, so you don't have to fix them all for this PR. It's okay to fix just this one, or plan to fix them all later in one fell swoop.

Comment (aabmass) [reply .circleci/config.yml:63]: 👍 good point

File: opentelemetry-exporter-gcp-trace/src/opentelemetry/exporter/cloud_trace/version.py [hunk 2]
```diff
@@ -12,4 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "0.18.dev0"
+__version__ = "0.19.dev0"
```
Comment (aabmass) [inline opentelemetry-exporter-gcp-trace/src/opentelemetry/exporter/cloud_trace/version.py:15]: Missed this in a previous PR somehow

Comment (punya) [reply opentelemetry-exporter-gcp-trace/src/opentelemetry/exporter/cloud_trace/version.py:15]: Do we have tooling to automate editing this and/or enforce consistency across the different packages?

**Pull request merged**